### ISSUE TITLE
Feature Request: API Token Resource and Data blocks

### ISSUE THREAD

**Issue by deepbrook** [philips-labs/terraform-provider-unleash#50]
Hello!

My team and I would like to request the following resource (and subsequently data blocks) to be added:

```hcl
data unleash_user "spam" {
  id = 1
}

resource "unleash_api_token" client" {
  username = data.unleash_user.username  # required
  type = "client"  // or "admin", "frontend" - required

  environment = "string"
  expires_at = "2055-01-01"
}

data unleash_api_token "other_client" {
  projects = [
    data.unleash_project.foo.name
  ]
  username = data.unleash_user.spam.username
}

```

### References
The relevant API docs can be found [here](https://docs.getunleash.io/reference/api/unleash/api-tokens)

**Comment by deepbrook** [comment]
There's also feedback on when to use "project" vs "projects": https://github.com/Unleash/unleash/issues/2189#issuecomment-1281999897

Essentially, `projects` should be used and its default value should be `[*]` (granting access to all projects)

**Comment by jdelucaa** [comment]
Hello @deepbrook 👋 
Your feature request is very welcome, let me have a look at this topic this week and I'll get back to you by the end of next week.
Thanks 😄

**Comment by jdelucaa** [comment]
@deepbrook can you elaborate a bit more on the data source and the query you would like to see executed? I'm assuming it's an `and`, so it will retrieve tokens that have the username specified AND **all projects** specified. Also, the query may return more than one entry.

**Comment by deepbrook** [comment]
@jdelucaa sure!

Yes, its supposed to be an `and`, mostly because - as you already stated- the endpoint may return several tokens.

Ideally, two data blocks are implemented: one which returns the API's result as is (in a result attribute, for example). This is useful to operate over several tokens if need be.

A second block would allow filtering for a single specific token, and raise an error if there is more than one token returned.

**Comment by jdelucaa** [comment]
@deepbrook yeah so, the API doesn't have any filters, it simply returns all api tokens, so I have to filter it myself based on the provided username and projects.

So you say that you want a data block that will return all API tokens with no filter at all and another one that would have filters and raise an error if there is more than one token?

For now I have both implemented, but the first one also have filters, the only diff is that the second one raises an error in case multiple tokens are returned.

**Comment by jdelucaa** [comment]
Available on v0.3.5  🚀🚀🚀
